### ISSUE TITLE
Demonstrate time aggregation in vignette

### ISSUE THREAD

**Issue by jacobbien** [cmu-delphi/epiprocess#24]
In the first half of the "aggregation.Rmd" vignette, we should include some examples of how to perform time aggregation.  In particular, the approach to demo is casting to `tsibble` on-the-fly and then leveraging the handy functions there.

A particularly useful example would involve adding aggregation to the epiweek level:

```
library(tsibble)
dat %>%
  as_tsibble(index = time_value, key = geo_value) %>%
  index_by(Epiweek = ~ epiweek(.)) %>%
  summarize(num_cases = sum(Cases))
```
Here the function `epiweek` function would have to be defined (perhaps based on the `MMWRweek` R package).

(For more context on this issue, see https://github.com/cmu-delphi/epitools/issues/7)

**Comment by ryantibs** [comment]
Thanks @jacobbien.  Wondering what @earowang thinks? 

We're currently thinking that casting to `tsibble` on-the-fly (which, as Jacob mentions, we'll demo in the first half of [this vignette](https://cmu-delphi.github.io/epitools/articles/aggregation.html)) is a good strategy to access all the `tsibble` utilities, without worrying about compatibility in all cases, up-front.  That is, it puts the onus on the user to define the `index` variable carefully when they want to use `tsibble` utilities, and not on us (packages designers) in general.  In general, it seems like how the `index` variable gets updated across various sliding and pivoting operations could get complicated.  

Do you see a downside to this approach, or have different perspectives on its pros & cons?

**Comment by earowang** [comment]
Coercing to `tsibble` as needful definitely works.

1. Do `index = time_value` and `key = geo_value` hold all the time? If so, users don't need to pass these parameters while coercing, by defining `as_tsibble.epi_tibble(x, ...)`.

```r
dat %>% 
  as_tsibble() %>% # less learning
  index_by()
```

2. {lubridate} also provides `epiweek()`, although not sure if they refer to the same epi weeks.

**Comment by qpmnguyen** [comment]
@ryantibs @jacobbien happy to take up writing the vignette for this issue if it's available.

**Comment by ryantibs** [comment]
@qpmnguyen Thanks! Actually, if you caught the latest discussion on slack, there may be some changes coming to the way we handle key variables in an `epi_df` object, and we may be (back to) considering having it inherit a `tsibble`. If so then that would affect this vignette (e.g., no need to cast to a `tsibble` if it already is one). 

I plan on Mon-Tues revisiting the discussion on slack and making a decision. If we do decide to move forward to make an `epi_df` a `tsibble`, then it's probably cleanest if you let me make this change before you attempt the vignette. If we don't, then you can just go for the vignette directly. Either way, let me get back to you early next week with a status update.

**Comment by ryantibs** [comment]
Closed by #37 #38.
